src/main/java/com/autotune/database/helper/DBConstants.java

```diff
@@ -15,9 +17,13 @@ public static final class SQLQUERY {
         public static final String SELECT_FROM_RESULTS_BY_EXP_NAME_AND_MAX_END_TIME = String.format("from KruizeResultsEntry k WHERE k.experiment_name = :%s and k.interval_end_time = (SELECT MAX(e.interval_end_time) FROM KruizeResultsEntry e  where e.experiment_name = :%s )", KruizeConstants.JSONKeys.EXPERIMENT_NAME, KruizeConstants.JSONKeys.EXPERIMENT_NAME);
         public static final String SELECT_FROM_RECOMMENDATIONS_BY_EXP_NAME = "from KruizeRecommendationEntry k WHERE k.experiment_name = :experimentName";
         public static final String SELECT_FROM_RECOMMENDATIONS_BY_EXP_NAME_AND_END_TIME = String.format("from KruizeRecommendationEntry k WHERE k.experiment_name = :%s and k.interval_end_time= :%s", KruizeConstants.JSONKeys.EXPERIMENT_NAME, KruizeConstants.JSONKeys.INTERVAL_END_TIME);
+        public static final String SELECT_FROM_RECOMMENDATIONS_BY_CLUSTER_NAME = "from KruizeRecommendationEntry k WHERE k.cluster_name = :clusterName and k.experiment_name IN :experimentList ORDER by interval_end_time DESC LIMIT :limit";
```

Comment:
Refrain from employing the 'limit' statement on its own...
Instead, employ the 'limit' statement in conjunction with carefully optimized 'where' conditions. The 'limit' statement processes the entire content and then returns a specified count. However, we can enhance efficiency by strategically utilizing 'where' conditions along with the 'limit' statement, thus mitigating the need to process the entire content.